src/main/java/com/faforever/client/coop/CoopController.java

```diff
@@ -125,6 +129,8 @@ public void initialize() {
     numberOfPlayersComboBox.getSelectionModel().select(0);
     numberOfPlayersComboBox.getSelectionModel().selectedItemProperty().addListener(observable -> loadLeaderboard());
 
+    leaderboardSearchTextField.textProperty().addListener((observable, oldValue, newValue) -> onSearchFieldChange(newValue));
```

Comment:
```suggestion
    leaderboardSearchTextField.textProperty().subscribe(this::onSearchFieldChange);
```
JavaFX has better syntax for this now

Reply:
Although really we can better set the predicate with a binding to the text field

Reply:
> Although really we can better set the predicate with a binding to the text field

I don't know how to do this. Would be helpful if you give an example.

Reply:
bindings are used to auto listen to changes in another property in javaFX. Coupled with the map and flatmap methods on observable values that are similar to those on the java streams they can create easy listener structures.

This is already done with a few other predicateProperty in the code so if you search for predicateProperty().bind you should find some examples. Then you just need a function here that takes the text and turns it into the desired predicate and can use that as `predicateProperty().bind(textProperty().map(this::createSearchPredicate))`